src/MusicSrc/MusicDevice.c

```diff
@@ -415,14 +430,208 @@ inline static void NativeMidiSendMessage(
     u.bData[3] = 0;
 
 //    INFO("NativeMidiSendMessage(): Sending MIDI data: 0x%08X", u.dwData);
-    const unsigned long err = midiOutShortMsg(outHandle, u.dwData);
+    const unsigned long err = midiOutShortMsg(*outHandle, u.dwData);
     if (err)
     {
         static char buffer[1024];
         midiOutGetErrorText(err, &buffer[0], 1024);
         WARN("NativeMidiSendMessage(): midiOutShortMsg() error: %s", &buffer[0]);
     }
 }
+#elif defined(__APPLE__)
+// inspired by rtmidi; see modified MIT license at https://github.com/thestk/rtmidi
+inline static CFStringRef EndpointName(const MIDIEndpointRef endpoint, const bool isExternal)
+{
+    CFMutableStringRef result = CFStringCreateMutable(NULL, 0);
+
+    // Begin with the endpoint's name.
+    CFStringRef str = NULL;
+    MIDIObjectGetStringProperty(endpoint, kMIDIPropertyName, &str);
+    if (str)
+    {
+        CFStringAppend(result, str);
+        CFRelease(str);
+        str = NULL;
+    }
+
+    // some MIDI devices have a leading space in endpoint name. trim
+    CFStringRef space = CFStringCreateWithCString(NULL, " ", kCFStringEncodingUTF8);
```

Comment:
You can use the macro `CFSTR()` here instead:

    CFStringRef space = CFSTR(" ");
    CFStringTrim(result, space);
    //CFRelease(space); // Has no effect on CFSTR-created CFStrings.

Or just

    CFStringTrim(result, CFSTR(" "));